repos/spack_repo/builtin/packages/py_rucio_clients/package.py

```diff
@@ -51,7 +52,8 @@ class PyRucioClients(PythonPackage):
     depends_on("py-packaging@24.2:", type=("build", "run"), when="@37:")
     depends_on("py-rich@13.7.1:", type=("build", "run"), when="@36:")
     depends_on("py-rich@13.9.4:", type=("build", "run"), when="@37:")
-    depends_on("py-typing-extensions@4.12.2:", type=("build", "run"))
+    depends_on("py-typing-extensions@4.12.2:", type=("build", "run"), when="@:37")
+    depends_on("py-typing-extensions@4.14.0:", type=("build", "run"), when="@38:")
```

Comment:
Rather than modifying previous lines with end ranges, it is more common to just add new lines (this reduces diff and blame noise):

```suggestion
    depends_on("py-typing-extensions@4.12.2:", type=("build", "run"))
    depends_on("py-typing-extensions@4.14.0:", type=("build", "run"), when="@38:")
```

This has the same effect of having a new version requirement for v38 and newer.